Fix the ordering guarantees claim. (#267)
dfinity/interface-spec: spec/index.md

### Ordering guarantees

> The order in which the various messages between canisters are delivered and executed is not fully specified. The guarantee provided by the IC is that successful function calls between two canisters are executed in order.

Reviewer: ```suggestion
The order in which the various messages between canisters are delivered and executed is not fully specified. The guarantee provided by the IC is that function calls between two canisters that start being executed do so in the order in which the corresponding messages were sent.
```

More precisely:

> -   Method calls between any *two* canisters, if delivered, start executing in order. Note that function call delivery can fail for arbitrary reasons (e.g., high system load).

Reviewer: ```suggestion
-   Messages between any *two* canisters, if delivered to the canister, start executing in order. Note that message delivery can fail for arbitrary reasons (e.g., high system load).
```
unifying the wording a bit

> -   Successful method calls between any *two* canisters are delivered in order. Note that function calls can fail for arbitrary reasons (e.g., high system load). Thus, canisters that need to ensure that two function calls are both executed, and moreover executed in a particular order, must wait for the completion of the first function call.

Reviewer: ```suggestion
-   Successful method calls between any *two* canisters are delivered in order. Note that function calls can fail for arbitrary reasons (e.g., high system load).
```

Reviewer: Discussed in the Interface spec meeting: there might be other ways for the canister to guarantee in-order execution, e.g., sequence numbers. CC @oggy-dfin

> -   Successful method calls between any *two* canisters are delivered in order. Note that function calls can fail for arbitrary reasons (e.g., high system load). Thus, canisters that need to ensure that two function calls are both executed, and moreover executed in a particular order, must wait for the completion of the first function call.

Reviewer: ```suggestion
-   Method calls between any *two* canisters start executing in order. Note that function calls can fail for arbitrary reasons (e.g., high system load). Thus, canisters that need to ensure that two function calls are both executed, and moreover executed in a particular order, must wait for the completion of the first function call.
```

Author: I don't quite like that suggestion because it suggests that calls will start executing (whereas one of them might not); but I think I understand your point, so I'll give it another shot; lmk what you think.
